### PR TITLE
docs: add third party to the specification of the declarative payment network

### PR DESCRIPTION
**Pull request by vrolland**
## Description of the changes

Add the third party (kind of referee) to the declarative payment network.

(N.B: I didn't rename YET the file to 0.2.0 to make the changes more readable)

I was wondering if we should add a new payment network for this even if the code and specs are very similar to the declarative one. Feel free to give your opinion.

### CODE REVIEW

File: packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md [hunk 1]
```diff
@@ -10,9 +10,10 @@ Prerequisite: Having read the advanced logic specification (see [here](/packages
 ## Description
 
 This extension allows to declare payments and the refunds in any currency.
-The payments and refunds are documented by the payer and the payee of the request.
 
-This extension do not ensure payment detection, only a consensus is made between the payer and the payee.
+The payments and refunds are documented by the payer, the payee of the request and optionnally a third party declared at the creation of the payment network.
+
+This extension do not ensure payment detection only a consensus is made between the payer and the payee or by a third party.
```
Comment (benjlevesque) [inline packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md:16]:
```suggestion
This extension does not ensure payment detection, only a consensus is made between the payer and the payee or by a third party.
```

File: packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md [hunk 2]
```diff
@@ -10,9 +10,10 @@ Prerequisite: Having read the advanced logic specification (see [here](/packages
 ## Description
```
Comment (benjlevesque) [inline packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md:10]: @vrolland, regarding your comment about the file name, I believe it should not contain a version at all, but include one at the top of the file maybe

File: packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md [hunk 3]
```diff
@@ -32,6 +33,7 @@ As a payment network, this extension allows to deduce a payment `balance` for th
 | **values.receivedRefundAmount**  | Amount | Amount of refund declared received             | **Mandatory** |
 | **values.paymentInstruction**    | String | Instruction to make payments                   | Optional      |
 | **values.refundInstruction**     | String | Instruction to make refunds                    | Optional      |
+| **values.thirdparty**    | [Identity](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/request-logic/specs/request-logic-specification.md#identity-role-and-signature) | Identity of the third party | Optional      |
```
Comment (benjlevesque) [inline packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md:36]:
```suggestion
| **values.thirdparty**    | [Identity](../request-logic/specs/request-logic-specification.md#identity-role-and-signature) | Identity of the third party | Optional      |
```

not tested, but I think it should work: https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-readmes#relative-links-and-image-paths-in-readme-files

File: packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md [hunk 4]
```diff
@@ -45,10 +47,11 @@ As a payment network, this extension allows to deduce a payment `balance` for th
 | --------------------------------- | ------ | ------------------------------------ | ------------- |
 | **id**                            | String | constant value: "pn-any-declarative" | **Mandatory** |
 | **type**                          | String | constant value: "paymentNetwork"     | **Mandatory** |
-| **version**                       | String | constant value: "0.1.0"              | **Mandatory** |
+| **version**                       | String | constant value: "0.2.0"              | **Mandatory** |
 | **parameters**                    | Object |                                      |               |
 | **parameters.paymentInstruction** | String | Instruction to make payments         | Optional      |
 | **parameters.refundInstruction**  | String | Instruction to make refunds          | Optional      |
+| **parameters.thirdparty**    | [Identity](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/request-logic/specs/request-logic-specification.md#identity-role-and-signature) | Identity of the third party | Optional      |
```
Comment (benjlevesque) [inline packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md:54]: use relative link

Comment (benjlevesque) [reply packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md:54]: shouldn't the anchor be `#identity` instead?

**Review by benjlevesque — APPROVED** [review]
nice! I would add a definition of the third party and maybe a concrete example of what a third party could be, as it might not be trivial.


I wonder if it would make sense to make `thirdparty` plural?  Could we have some use cases where there would be several actors able to declare payments?

**Review by siibars — APPROVED** [review]

**Review by Dadogg80 — APPROVED** [review]

**Review by yomarion — APPROVED** [review]
Nice and easy

File: packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md (hunk 4 above)
Comment (vrolland) [reply packages/advanced-logic/specs/payment-network-any-declarative-0.1.0.md]: Well, this anchor is giving a better picture of what is an identity.

**Pull request merged**